tsfresh/utilities/dataframe_functions.py

```diff
@@ -440,57 +528,76 @@ def roll_time_series(df_or_dict, column_id, column_sort, column_kind, rolling_di
     else:
         grouper = [column_id, ]
 
-    if column_sort is not None and df[column_sort].dtype != np.object:
-
+    if column_sort is not None:
         # Require no Nans in column
         if df[column_sort].isnull().any():
             raise ValueError("You have NaN values in your sort column.")
 
         df = df.sort_values(column_sort)
 
-        # if rolling is enabled, the data should be uniformly sampled in this column
-        # Build the differences between consecutive time sort values
+        if df[column_sort].dtype != np.object:
+            # if rolling is enabled, the data should be uniformly sampled in this column
+            # Build the differences between consecutive time sort values
 
-        differences = df.groupby(grouper)[column_sort].apply(
-            lambda x: x.values[:-1] - x.values[1:])
-        # Write all of them into one big list
-        differences = sum(map(list, differences), [])
-        # Test if all differences are the same
-        if differences and min(differences) != max(differences):
-            warnings.warn("Your time stamps are not uniformly sampled, which makes rolling "
-                          "nonsensical in some domains.")
+            differences = df.groupby(grouper)[column_sort].apply(
+                lambda x: x.values[:-1] - x.values[1:])
+            # Write all of them into one big list
+            differences = sum(map(list, differences), [])
+            # Test if all differences are the same
+            if differences and min(differences) != max(differences):
+                warnings.warn("Your time stamps are not uniformly sampled, which makes rolling "
+                              "nonsensical in some domains.")
 
     # Roll the data frames if requested
     rolling_direction = np.sign(rolling_direction)
 
     grouped_data = df.groupby(grouper)
-    max_timeshift = max_timeshift or grouped_data.count().max().max()
+    prediction_steps = grouped_data.count().max().max()
 
-    if np.isnan(max_timeshift):
+    if np.isnan(prediction_steps):
         raise ValueError("Somehow the maximum length of your time series is NaN (Does your time series container have "
                          "only one row?). Can not perform rolling.")
+    max_timeshift = max_timeshift or prediction_steps
+
+    # Todo: not default for columns_sort to be None
+    if column_sort is None:
+        df["sort"] = range(df.shape[0])
```

Comment:
don't you have to set `columns_sort` to `sort` here?

Reply:
I handled this case in the `_roll_out_time_series` separately. Because the sort column is an increasing number throughout all `id`s and `kind`s in this case, the index build from the old `id` and the `sort` column does not make much sense. Example:

You fill in data like
```
id  value
1  a
1  b
1  c
2  d
2  e
```
and the added sort column will be
```
id  value sort
1  a        1
1  b        2
1  c        3
2  d        4
2  e         5
```

If I would use that value, we would end up with some ids names `id=2,timeshift=5`, which would not make much sense (in my opinion). Rather, they are now named `id=2,timeshift=0` and `id=2,timeshift=1` .